flocker/node/test/test_loop.py

```diff
@@ -724,6 +726,43 @@ def test_status_update_while_sleeping_no_discovery(self):
             remaining_discover_calls - len(self.deployer.local_states),
             0)
 
+    def test_longer_sleep_when_converged(self):
+        """
+        When a convergence loop results in a ``NoOp`` the sleep is based on
+        that configured on the ``Deployer``.
+        """
+        loop = self.convergence_iteration(initial_action=NoOp(),
+                                          later_actions=[NoOp(), NoOp()])
+
+        # Another update received while sleeping:
+        loop.receive(_ClientStatusUpdate(
+            client=self.make_amp_client([self.local_state]),
+            configuration=self.configuration, state=self.cluster_state))
+
+        # No additional discovery done due to update:
+        pre_sleep = len(self.deployer.local_states)
+
+        # Sleep until right before lowest possible random sleep interval:
+        self.reactor.advance(
+            self.deployer.poll_interval * (1 - ConvergenceLoop._JITTER_RANGE)
+            - 0.05)
+        mid_sleep = len(self.deployer.local_states)
+        # Sleep until right after highest possible sleep interval
+        self.reactor.advance(
+            self.deployer.poll_interval + (2 * ConvergenceLoop._JITTER_RANGE)
+            + 0.05
+            # Extra bit of sleep to ensure we don't break on floating
+            # point rounding errors:
+            + 0.00001)
```

Comment:
That still sounds to me like it would at least make sense to subtract some tiny value.  But I don't know what the 0.05 is for, maybe that already accounts for this.